### PR TITLE
Closes #9433: Update Flank version in ui-tests Dockerfile

### LINKED ISSUE #9433: Update Flank version in ui-tests Dockerfile
Flank (master) [supports a new server side sharding API which is faster](https://firebase-community.slack.com/archives/C1MTSQ5QT/p1585235777017600?thread_ts=1585235665.017500&cid=C1MTSQ5QT).
https://github.com/Flank/flank/tree/flank_snapshot in 

https://github.com/mozilla-mobile/fenix/blob/73a7330b2b002908e6f2bb30c9fe23fe6b261160/taskcluster/docker/ui-tests/Dockerfile#L13 

From the Firebase Community Slack re: flank_snapshot pre-release:

_"In Flank stable, the matrices are executed in parallel however they're polled for status updates sequentially.  In the latest version, there's only one matrix that contains all the shards which helps with performance and logging."_

_"I think the snapshot is stable, all the tests are passing. We're planning a big release in the next few months."_

(CC @rpappalax @JohanLorenzo)

### PR DESCRIPTION
**Pull request by AaronMT**
v8.0.1 -> flank_snapshot (see #9433)

https://github.com/Flank/flank/releases

### CODE REVIEW

**Comment by AaronMT** [comment]
122 tests / 122 shards

```[task 2020-03-27T18:39:53.416Z] Request failed, retrying 1x com.google.api.client.googleapis.json.GoogleJsonResponseException: 400 Bad Request
[task 2020-03-27T18:39:53.418Z] {
[task 2020-03-27T18:39:53.418Z]   "code" : 400,
[task 2020-03-27T18:39:53.419Z]   "errors" : [ {
[task 2020-03-27T18:39:53.419Z]     "domain" : "global",
[task 2020-03-27T18:39:53.419Z]     "message" : "num_shards must be >= 1 and <= 50.",
[task 2020-03-27T18:39:53.419Z]     "reason" : "badRequest"
[task 2020-03-27T18:39:53.419Z]   } ],
[task 2020-03-27T18:39:53.419Z]   "message" : "num_shards must be >= 1 and <= 50.",
[task 2020-03-27T18:39:53.420Z]   "status" : "INVALID_ARGUMENT"
[task 2020-03-27T18:39:53.420Z] }
```

Interesting change. Should we try 50?

File: taskcluster/ci/ui-test/kind.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ jobs:
             commands:
                 - [wget, {artifact-reference: '<signing/public/build/x86/geckoNightly/target.apk>'}, '-O', app.apk]
                 - [wget, {artifact-reference: '<signing-android-test/public/build/noarch/geckoNightly/target.apk>'}, '-O', android-test.apk]
-                - [automation/taskcluster/androidTest/ui-test.sh, x86, app.apk, android-test.apk, '-1']
+                - [automation/taskcluster/androidTest/ui-test.sh, x86, app.apk, android-test.apk, '50']
```
Comment (JohanLorenzo) [inline taskcluster/ci/ui-test/kind.yml]: Quick question: does 50 here overrides what's defined in `automation/taskcluster/androidTest/flank-x86.yml`? (not blocking this PR)

Comment (AaronMT) [reply taskcluster/ci/ui-test/kind.yml]: Yeah we should always use the shard value provided by the templates, which is not happening right now https://github.com/mozilla-mobile/fenix/issues/9185

**Comment by AaronMT** [comment]
```7m  4s``` on flank_snapshot

I'm also looking into the ```GoogleJsonResponseException``` on the reporting at the end.

**Comment by AaronMT** [comment]
@JohanLorenzo Just want to confirm wether new flank-snapshot changes (that could update reguarily) will force new docker rebuilding and if so, is that ok?

**Comment by JohanLorenzo** [comment]
@AaronMT I'm sorry, it won't. Some changes have to happen in `taskcluster/docker/ui-tests/Dockerfile` in order to force a new docker image build. It could be dummy change like adding a space somewhere. The original intent is to build new images only when it's needed and the only way for taskgraph to know is to monitor changes in checked-in files.

That said, we could write a cron job that monitors https://api.github.com/repos/Flank/flank/releases and then opens a PR to update `taskcluster/docker/ui-tests/Dockerfile`. I solved a similar case 2 weeks ago in https://github.com/mozilla-mobile/fenix/pull/9251 🙂

**Pull request merged**